repository_service_tuf_api/api/targets.py

```diff
@@ -48,3 +48,23 @@ def delete(
     _user=Security(validate_token, scopes=[SCOPES_NAMES.delete_targets.value]),
 ):
     return targets.delete(payload)
+
+
+@router.post(
+    "/publish/",
```

Comment:
I see that not all of the methods have this attribute set to anything different than `/`.
Am I right in my observation that the endpoint names are constructed like that if they are an f-string:
`f"api/{routert_tag}/{router_prefix}/{method_suffix}"` ?

Reply:
The `app.py` defines the `rstuf_app` "/"
https://github.com/vmware/repository-service-tuf-api/blob/3392117cdca4488158b02958e1601796f8766b0a/app.py#L17-L20

The `api_v1` also defines the "/api/v1/"

https://github.com/vmware/repository-service-tuf-api/blob/3392117cdca4488158b02958e1601796f8766b0a/app.py#L23

- What allows you to have multiple API versions (v2, v3...) to keep the backward compatibility with old versions of API Clients

In the `repository_service_tuf_api/api/` we implement the routes, enabling us to have multiple implementations if needed and import them with specific versions or multiple

https://github.com/vmware/repository-service-tuf-api/blob/3392117cdca4488158b02958e1601796f8766b0a/app.py#L10-L15

Reply:
> * What allows you to have multiple API versions (v2, v3...) to keep the backward compatibility with old versions of API Clients
> 
> In the `repository_service_tuf_api/api/` we implement the routes, enabling us to have multiple implementations if needed and import them with specific versions or multiple
> 
> https://github.com/vmware/repository-service-tuf-api/blob/3392117cdca4488158b02958e1601796f8766b0a/app.py#L10-L15

Cool, thanks for mentioning that part as well.

Reply:
So, the naming is 
`f"{api_v1.prefix}/{router_prefix}{method_suffix}"`?

Then, what is the role of the `tags` and `response` attributes for the APIRouter?
It seems they are not used in the naming.

Reply:
> So, the naming is `f"{api_v1.prefix}/{router_prefix}{method_suffix}"`?

`f"{api_v1.prefix}/{router.prefix}/{router.<method>.path}"`

> Then, what is the role of the `tags` and `response` attributes for the APIRouter? It seems they are not used in the naming.

Tags are used in the Documentation. The same router can be part of different tags, so you can reuse the same router in multiple API prefixes. It is more on the[ FastAPI framework details.](https://fastapi.tiangolo.com/advanced/)

Simulation.
![Screenshot 2023-02-03 at 12 46 42](https://user-images.githubusercontent.com/546882/216595848-8dc4225d-b3d8-4423-9cec-583f08c737f4.png)

The responses are also related to swagger documentation.
https://fastapi.tiangolo.com/tutorial/response-status-code/
https://fastapi.tiangolo.com/advanced/additional-responses/